### PR TITLE
Switch to uppercase environment variables for AWS (#3438)

### PR DESCRIPTION
**Pull request by alecpm**
## Description
When setting environment variables on Heroku or in Docker containers uses an all uppercase spelling of the AWS environment variables for better compatibility with recent boto versions.

## Motivation and Context
See #3438

## How Has This Been Tested?
Automated tests have been updated and run.

### CODE REVIEW

File: dallinger/command_line/docker.py [hunk 1]
```diff
@@ -208,9 +208,9 @@ def deploy_image(image, mode, config_options):
     dashboard_password = secrets.token_urlsafe(8)
     dallinger_uid = str(uuid.uuid4())
     config_dict = {
-        "aws_access_key_id": config.get("aws_access_key_id"),
-        "aws_secret_access_key": config.get("aws_secret_access_key"),
-        "aws_region": config.get("aws_region"),
+        "AWS_ACCESS_KEY_ID": config.get("aws_access_key_id"),
+        "AWS_SECRET_ACCESS_KEY": config.get("aws_secret_access_key"),
+        "AWS_REGION": config.get("aws_region"),
```
Comment (silviot) [inline dallinger/command_line/docker.py:213]: I believe it should be `AWS_DEFAULT_REGION`. [See boto docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables).

**Comment by silviot** [comment]
I tried to replicate the test failure locally but I was not able to.

**Comment by alecpm** [comment]
@silviot same, after review @jessesnyder advised removing the test which doesn't actually test anything useful or do what it claims to do. Good catch on `AWS_DEFAULT_REGION` I was making the bad assumption that boto had switched `aws` cli variables, but apparently `aws` cli deprecated `AWS_REGION` and prefers `AWS_DEFAULT_REGION` too! I'll make that change shortly.

**Comment by alecpm** [comment]
@silviot OK, I think this is all set. Tests are passing and it also now pulls in AWS values to the experiment config from the uppercase (and slightly changed) environment variables.

**Review by jessesnyder — APPROVED** [review]
LGTM

**Pull request merged**